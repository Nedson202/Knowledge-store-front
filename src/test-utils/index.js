import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux';

// eslint-disable-next-line react/prop-types
export const AllProviders = ({ children, customStore }) => (
  <MockedProvider removeTypename>
    <Provider store={customStore || store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  </MockedProvider>
);

export * from '@testing-library/react';
