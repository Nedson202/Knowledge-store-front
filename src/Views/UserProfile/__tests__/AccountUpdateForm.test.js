import React from 'react';

import { AllProviders, render } from 'test-utils';

import AccountUpdateForm from '..';

describe('AccountUpdateForm', () => {
  it('should render form labels', () => {
    const {
      container, asFragment, getByText, getAllByText
    } = render(
      <AllProviders>
        <AccountUpdateForm />
      </AllProviders>
    );

    getAllByText('Username');
    getByText('Email address');

    expect(asFragment()).toMatchSnapshot();
    expect(container).toBeVisible();
  });

  it('should render action buttons', () => {
    const { getByText } = render(
      <AllProviders>
        <AccountUpdateForm />
      </AllProviders>
    );

    getByText('Update Profile');
  });
});
