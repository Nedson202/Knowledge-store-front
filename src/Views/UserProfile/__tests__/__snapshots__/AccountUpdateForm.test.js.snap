// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUpdateForm should render form labels 1`] = `
<DocumentFragment>
  <div
    class="container-content"
  >
    <div
      class="profile-details"
    >
      <div
        class="profile-details__image"
      >
        <img
          alt="Avatar"
          height="200px"
          id="show-image"
          width="200px"
        />
      </div>
      <div
        class="profile-details__info"
      >
        <span>
          Username
        </span>
        <h4 />
        <span>
          Email
        </span>
        <h4 />
      </div>
    </div>
    <div
      class="profile-update"
    >
      <div
        class="profile-upload"
      >
        <label
          class="btn btn-primary bt-raised rounded-circle"
          id="image-upload"
        >
          <div>
            <div
              id="dropzone"
            >
              <ion-icon
                class="text-center"
                name="image"
              />
              <p>
                Select image
              </p>
            </div>
          </div>
          <input
            hidden=""
            name="image"
            type="file"
          />
        </label>
        <div
          class="update-form"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                class="bmd-label-floating"
                for="profile-name"
              >
                Username
              </label>
              <input
                class="form-control"
                id="profile-username"
                name="username"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="bmd-label-floating"
                for="profile-email"
              >
                Email address
              </label>
              <input
                class="form-control"
                id="profile-email"
                name="email"
                type="email"
                value=""
              />
            </div>
            <div
              class="action-buttons"
            >
              <button
                class="btn btn-raised text-case"
                disabled=""
                id="update-button"
                type="button"
              >
                Update Profile
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="password-update"
      >
        <form>
          <div
            class="form-group"
          >
            <label
              class="bmd-label-floating"
              for="old-password"
            >
              Old Password
            </label>
            <input
              autocomplete="new-password"
              class="form-control"
              id="old-password"
              name="oldPassword"
              type="password"
              value=""
            />
            <div
              data-testid="old-password-icon"
              id="old-password-icon"
              role="button"
              tabindex="0"
            >
              <ion-icon
                class="hide"
                id="old-password-remove-hide"
                name="eye-off"
              />
              <ion-icon
                id="old-password-add-hide"
                name="eye"
              />
            </div>
          </div>
          <div
            class="form-group"
          >
            <label
              class="bmd-label-floating"
              for="new-password"
            >
              New Password
            </label>
            <input
              autocomplete="new-password"
              class="form-control"
              id="new-password"
              name="newPassword"
              type="password"
              value=""
            />
            <div
              data-testid="new-password-icon"
              id="new-password-icon"
              role="button"
              tabindex="0"
            >
              <ion-icon
                class="hide"
                id="new-password-remove-hide"
                name="eye-off"
              />
              <ion-icon
                id="new-password-add-hide"
                name="eye"
              />
            </div>
          </div>
          <div
            class="form-group"
          >
            <button
              class="btn btn-raised text-case"
              id="update-password"
              type="button"
            >
              Update Password
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
